### PR TITLE
 Added support for returning of generator

### PR DESCRIPTION
**Pull request by maciejs**

### CODE REVIEW

**Comment by ssbarnea** [comment]
I am wondering if it would not be better to make ResultList a generator instead. If possible this would be preferable because it would allow us to even switch the default behavior in the future without breaking existing code.

**Comment by maciejs** [comment]
Sounds reasonable. This could be implemented as next step.

**Comment by hdost** [comment]
@ssbarnea is your thought that this change would be required instead of the one here?

**Comment by ssbarnea** [comment]
Mainly I would propose updating this in such way that it would not need a new parameter. ResultList should be a generator.

Making pytho-api more complex will only make it harder to use, test and refactor.

As is python it does not have to be either one or another, it can be both.

File: jira/client.py [hunk 1]
```diff
@@ -374,6 +377,7 @@ def __init__(self,
                  proxies=None,
                  timeout=None,
                  auth=None,
+                 generator=False,
```
Comment (ssbarnea) [inline jira/client.py]: Please make ResultList act as a generator instead of adding a new parameter and different return values.

**Pull request closed without merging**